### PR TITLE
cmd: Use genericclioptions

### PR DESCRIPTION
**Pull request by mauriciovasquezbernal**
genericclioptions is helper package that makes it easier to develop plugins
that offer a similar experience to kubectl.

This commit uses this package, the main change is that the custom handling of
the namespace is removed everywhere. As a result of this, commands that doesn't
specify a namespace act on the configured namespace in the kubeconfig file.

Part of https://github.com/kinvolk/inspektor-gadget/issues/98.

### CODE REVIEW

**Comment by alban** [comment]
Does it also work for in-cluster kubeconfig?

**Comment by mauriciovasquezbernal** [comment]
> Does it also work for in-cluster kubeconfig?

Yes, it does. I tried it by running `kubectl-gadget` inside the gadget container.

**Comment by alban** [comment]
Could we have a `-A` / `--all-namespaces` option? (there is one in traceloop but not the other bcc-based gadgets, as far as I understand)

For the `profile` gadget, it would be useful to be able to trace on the host. I think this patch removes that possibility? Could we have a `--host` option? I guess that only makes sense for the `profile` gadget.

**Comment by mauriciovasquezbernal** [comment]
I pushed a change to support `-A, --all-namespaces` option. Before this PR it was not possible to track events on the host, it can be easily implemented. I'll open an issue for that.

I also did some small changes to the flags, added short options `-l`, `-c`, and `-p` and renamed `--label` to `--selector` to make it similar to `kubectl`. Please let me know what do you think about it.

File: cmd/kubectl-gadget/bcck8s.go [hunk 1]
```diff
@@ -97,18 +95,51 @@ func init() {
...
-				args[i],
-				"",
-				fmt.Sprintf("Kubernetes %s selector", args[i]))
-		}
+		command.PersistentFlags().StringVarP(
+			&labelParam,
+			"selector",
+			"l",
+			"",
+			fmt.Sprintf("Labels selector to filter on. Only '=' is supported.(e.g. key1=value1)."),
```
Comment (alban) [inline cmd/kubectl-gadget/bcck8s.go:107]: `s/.(/ (/`

Comment (alban) [reply cmd/kubectl-gadget/bcck8s.go:107]: You could add two labels in the example to show they are separated by `,`:
`-l key1=value1,key2=value2`

File: cmd/kubectl-gadget/main.go [hunk 2]
```diff
@@ -2,25 +2,26 @@ package main
...
 	cobra.OnInitialize(cobraInit)
-
-	// add kubeconfig flag
-	rootCmd.PersistentFlags().String(
-		"kubeconfig",
-		"",
-		"Path to kubeconfig file")
-	viper.BindPFlag("kubeconfig", rootCmd.PersistentFlags().Lookup("kubeconfig"))
+	KubernetesConfigFlags.AddFlags(rootCmd.PersistentFlags())
+	viper.SetEnvKeyReplacer(strings.NewReplacer("-", "_"))
```
Comment (alban) [inline cmd/kubectl-gadget/main.go:24]: Do we have any flag with a `-` or environment variable with a `_`?

File: cmd/kubectl-gadget/bcck8s.go (hunk 1 above)
Comment (mauriciovasquezbernal) [reply cmd/kubectl-gadget/bcck8s.go:107]: Done!

File: cmd/kubectl-gadget/main.go (hunk 2 above)
Comment (mauriciovasquezbernal) [reply cmd/kubectl-gadget/main.go:24]: There are some flags from the `genericclioptions` package with `-` like `as-group`, `certificate-authority`, `client-certificate`, etc.
It was suggested to be used in the plugin example.

**Pull request merged**